packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-target-group.ts

```diff
@@ -111,8 +127,16 @@ export class ApplicationTargetGroup extends TargetGroupBase implements IApplicat
       if (props.slowStart !== undefined) {
         this.setAttribute('slow_start.duration_seconds', props.slowStart.toSeconds().toString());
       }
-      if (props.stickinessCookieDuration !== undefined) {
-        this.enableCookieStickiness(props.stickinessCookieDuration);
+      if (props.stickinessCookieName !== undefined) {
+        if (props.stickinessCookieName.startsWith('AWSALB') || props.stickinessCookieName.startsWith('AWSALBAPP') || props.stickinessCookieName.startsWith('AWSALBTG')) {
+          throw new Error('App cookie names that start with the following prefixes are not allowed: AWSALB, AWSALBAPP, and AWSALBTG; they\'re reserved for use by the load balancer.');
+        }
+        if (props.stickinessCookieName === '') {
+          throw new Error('App cookie name cannot be an empty string.');
+        }
+      }
```

Comment:
@njlynch Should I move `stickinessCookieName` checks to `enableCookieStickiness` method since that method can be invoked elsewhere?

i.e. 
```ts
    // listener.test.ts
    const group = listener.addTargets('Group', {
      port: 80,
      targets: [new FakeSelfRegisteringTarget(stack, 'Target', vpc)],
    });
    group.enableCookieStickiness(cdk.Duration.hours(1), 'MyDeliciousCookie');
```

Reply:
Yes, that sounds like the right thing to do.